### ISSUE TITLE
Issue with saving/loading torch format objects when building targets via ssh

### ISSUE THREAD

**Issue by mattwarkentin** [wlandau/targets#207]
## Prework

* [x] Read and agree to the [code of conduct](https://github.com/wlandau/targets/blob/main/CODE_OF_CONDUCT.md) and [contributing guidelines](https://github.com/wlandau/targets/blob/main/CONTRIBUTING.md).
* [x] If there is [already a relevant issue](https://github.com/wlandau/targets/issues), whether open or closed, comment on the existing thread instead of posting a new issue.
* [x] Post a [minimal reproducible example](https://www.tidyverse.org/help/) so the maintainer can troubleshoot the problems you identify. A reproducible example is:
    * [x] **Runnable**: post enough R code and data so any onlooker can create the error on their own computer.
    * [x] **Minimal**: reduce runtime wherever possible and remove complicated details that are irrelevant to the issue at hand.
    * [x] **Readable**: format your code according to the [tidyverse style guide](https://style.tidyverse.org/).

## Description

Hi @wlandau,

So I have started to use `torch` quite a bit more and I have noticed that `format = "torch"` works differently when ran locally, versus ran remotely. Something about the serialization doesn't seem to make it through the sockets in a successful manner. This may not actually be a `targets` problem, it may rest on the shoulders of `torch_save()` and `torch_load()` which are currently somewhat experimental. Probably should loop @dfalbel in.

If I build my torch model "locally" with `tar_make()` or `tar_make_clustermq()`/`options(clustermq.scheduler="multiprocess")`, then I am able to load the `torch` model from storage and it seems entirely intact.

However, when I run `tar_make_clustermq()`/`options(clustermq.scheduler="ssh")`, the model makes it back to storage and is seemingly loadable, but some of the `R6` fields that contained tensors are broken (e.g. `fit$parameters`, which is the model weights/biases).

## Reproducible example

```r
library(targets)

pkgs <- c(
  "tidyverse",
  "torch"
)

options(
  clustermq.scheduler = "multiprocess"
  #clustermq.scheduler = "ssh",
  #clustermq.ssh.host = "USER@HOST"
)

tar_option_set(
  packages = pkgs,
  error = "stop",
  memory = "persistent",
  retrieval = "main",
  storage = "main",
  deployment = "worker",
  garbage_collection = TRUE
)

fit_model <- function() {
  d_in <- 3
  d_out <- 1
  n <- 100

  x <- torch_randn(n, d_in)
  y <- x[, 1, NULL] * 0.2 - x[, 2, NULL] * 1.3 - x[, 3, NULL] * 0.5 + torch_randn(n, 1)

  ### define the network ----

  # dimensionality of hidden layer
  d_hidden <- 32

  model <- nn_sequential(
    nn_linear(d_in, d_hidden),
    nn_relu(),
    nn_linear(d_hidden, d_out)
  )

  ### network parameters ----

  learning_rate <- 0.08

  optimizer <- optim_adam(model$parameters, lr = learning_rate)

  ### training loop ----

  for (t in 1:200) {

    ### -------- Forward pass --------
    y_pred <- model(x)

    ### -------- compute loss --------
    loss <- nnf_mse_loss(y_pred, y, reduction = "sum")
    if (t %% 10 == 0)
      cat("Epoch: ", t, "   Loss: ", loss$item(), "\n")

    ### -------- Backpropagation --------
    optimizer$zero_grad()
    loss$backward()

    ### -------- Update weights --------
    optimizer$step()
  }
  return(model)
}

tar_pipeline(
  tar_target(
    fit,
    fit_model(),
    format = "torch"
  )
)
```

## Expected result

When ran locally, `fit$parameters` should print a list of tensors of model weights/biases. When ran remotely via ssh, the following error is produced:
```
$`0.weight`
torch_tensor 
Error in cpp_torch_tensor_print(self$ptr) : 
  argument "n" is missing, with no default
```
This is the same error that was produced when I forgot to set `format="torch"` for that target and the stored object was broken. So I believe it has to do with the serialization. Just off that it varies based on ssh vs. local.

**Comment by mattwarkentin** [comment]
Relevant to mlverse/torch#150

**Comment by mattwarkentin** [comment]
Updating to the dev version of `torch` inside the container seems to have fixed the issue. Sorry about that.

**Comment by dfalbel** [comment]
Glad it was only a version mismatch @mattwarkentin ! The problem here was that the tensor saved with an older version of torch was using `cpp_tensor_print` expecting to have a single argument ( this is saved in the R6 class). however in newest versions of torch `cpp_tensor_print` that expects 2 arguments. A solution would be to recreate the tensor with a newest version of the class. Something like: `torch::Tensor$new(ptr = t$ptr)`.

**Comment by mattwarkentin** [comment]
Thanks for the suggestion, @dfalbel.
